src/app/nextversion/nextversion.go

```diff
@@ -146,6 +154,12 @@ func NextVersion(cCtx *cli.Context) error {
 
 	case nextOverride == nil && next != nil:
 		nextStr = next.String()
+		if err != nil {
+			if cCtx.Bool("fail") {
+				return cli.Exit(fmt.Errorf("bumping source: %w", err), 1)
+			}
+			log.Warnf("None of the changelog entries produced a version bump, returning current version")
+		}
```

Comment:
I think nesting this much makes the code a bit unreadable (mostly because of this big ugly switch case I introduced 😓)

We should probably refactor it at this point. For now, could we make this error check specific `errors.Is` and add a comment for our future selves?

```suggestion
		// If we don't have an override, the bumper did not bump anything, and the user has set `--fail`, then we should error out.
		if errors.Is(err, ErrNoNewVersion)  {
			log.Warnf("None of the changelog entries produced a version bump, returning current version")
			if cCtx.Bool("fail") {
				return cli.Exit(fmt.Errorf("failing by user request: %w", err), 1)
			}
		}
```